### PR TITLE
clean: Use sbt-native-packager to build native-image

### PR DESCRIPTION
**Pull request by josemiguelmelo**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -50,3 +50,19 @@ scmInfo := Some(
 
 fork in Test := true
 cancelable in Global := true
+
+enablePlugins(GraalVMNativeImagePlugin)
+graalVMNativeImageGraalVersion := Some("19.3.0-java8")
```
Comment (lolgab) [inline build.sbt]: Update graalvm here

File: .circleci/config.yml [hunk 2]
```diff
@@ -94,26 +108,23 @@ workflows:
...
           cmd: |
-            sbt "assembly"
-
-            ./scripts/publish-native.sh -n codacy-coverage-reporter -m com.codacy.CodacyCoverageReporter -t docker $(cat .version)
-
-            mkdir -p ~/workdir/artifacts/
-            mv codacy-coverage-reporter-linux-$(cat .version) ~/workdir/artifacts/
-            mv ~/workdir/target/codacy-coverage-reporter-assembly-$(cat .version).jar ~/workdir/artifacts/
+            sbt assembly
+            sbt 'show graalvm-native-image:packageBin'
```
Comment (lolgab) [inline .circleci/config.yml]:
```suggestion
            sbt graalvm-native-image:packageBin
```

Comment (lolgab) [reply .circleci/config.yml]: I don't think you need assembly

Comment (lolgab) [reply .circleci/config.yml]: If memory is enough you don't have to pay sbt startup time twice

File: build.sbt (hunk 1 above)
Comment (josemiguelmelo) [reply build.sbt]: ✅

File: .circleci/config.yml [hunk 3]
```diff
@@ -57,6 +57,20 @@ jobs:
...
+  package_artifacts:
+    docker:
+      - image: hairyhenderson/upx:latest
+    working_directory: ~/workdir
+    steps:
+      - run: apk add ca-certificates
+      - attach_workspace:
+          at: ~/workdir
+      - run: |
+          upx --lzma -o codacy-coverage-reporter-compressed ~/workdir/target/graalvm-native-image/codacy-coverage-reporter
```
Comment (andreaTP) [inline .circleci/config.yml]:
```
mkdir -p ~/workdir/artifacts/
cp ~/workdir/target/codacy-coverage-reporter-assembly-$(cat .version).jar ~/workdir/artifacts/
upx --lzma -o codacy-coverage-reporter-compressed ~/workdir/artifacts/codacy-coverage-reporter-linux-$(cat .version)
```

WDYT?

File: .circleci/config.yml [hunk 4]
```diff
@@ -94,26 +108,23 @@ workflows:
...
+          name: create_artifacts
           cmd: |
-            sbt "assembly"
-
-            ./scripts/publish-native.sh -n codacy-coverage-reporter -m com.codacy.CodacyCoverageReporter -t docker $(cat .version)
-
-            mkdir -p ~/workdir/artifacts/
-            mv codacy-coverage-reporter-linux-$(cat .version) ~/workdir/artifacts/
-            mv ~/workdir/target/codacy-coverage-reporter-assembly-$(cat .version).jar ~/workdir/artifacts/
+            sbt assembly
```
Comment (andreaTP) [inline .circleci/config.yml]: you can squash the two commands into one single sbt invocation.

File: .circleci/config.yml [hunk 5]
```diff
@@ -57,6 +57,20 @@ jobs:
           name: Publish final orb
           command: circleci orb publish orb.yml codacy/coverage-reporter@$(cat .version) --token $CIRCLE_TOKEN
 
+  package_artifacts:
+    docker:
+      - image: hairyhenderson/upx:latest
```
Comment (andreaTP) [inline .circleci/config.yml]: please avoid using moving tags.

File: .circleci/config.yml (hunk 3 above)
Comment (josemiguelmelo) [reply .circleci/config.yml]: Nice suggestion, it worked 🙏

File: .circleci/config.yml (hunk 4 above)
Comment (josemiguelmelo) [reply .circleci/config.yml]: Done ✅

File: .circleci/config.yml (hunk 5 above)
Comment (josemiguelmelo) [reply .circleci/config.yml]: You're right! Set a specific image version ✅

**Comment by lolgab** [comment]
Next is to port this to every GraalVM binary we have!

**Pull request merged**